makefiles/buildtests.inc.mk

```diff
@@ -255,7 +255,7 @@ info-boards-features-missing:
 
 FEATURES_REQUIRED += $(FEATURES_OPTIONAL)
 
-ifneq (, $(filter info-boards-supported info-boards-features-missing info-build, $(MAKECMDGOALS)))
+ifneq (, $(filter buildtest info-boards-supported info-boards-features-missing info-build info-buildsizes info-buildsizes-diff, $(MAKECMDGOALS)))
```

Comment:
@miri64 the code below generate `${BOARDS}` but this `ifneq` guards the code to be only exec for specific make targets, e.g. for `info-boards-supported` but w/o this PR not for target `buildtest` which IMHO make no sense. So I added all targets that require the boards list here and that way the recursive call of `make info-boards-supported` is obsolete which thereby also fixes described issues with the exported features variables, see e.g. #6690